src/uri.rs

```diff
@@ -389,6 +389,48 @@ impl StdError for UriError {
     }
 }
 
+// TODO: Is ByteStr adding value? Inline it?
```

Comment:
tldr: Im very happy to leave it alone and remove the comment for now.

I was playing with it last night, what I found was that more or less we do not interact with `ByteStr` or `Bytes` mostly we interact with `&str`. But at the same time we don't know the lifetime.

What would be ideal is if we could store `ByteStr {Bytes, &str /* pointing to Bytes */}`. Then if Uri was only using a `&str`, `ByteStr` would not have much use.

Thoughts?